src/crypto/sign.rs

```diff
@@ -1,28 +1,32 @@
 use std::fmt;
-use std::hash::Hash;
+use std::hash::{Hash, Hasher};
 
 use super::BincodeError;
 
 use bincode::serialize_into;
 
-use serde::{Deserialize, Serialize};
-
-use snafu::{ensure, Backtrace, ResultExt, Snafu};
-
-use sodiumoxide::crypto::sign::{
-    gen_keypair, sign_detached, verify_detached, PublicKey as SodiumPublicKey,
-    SecretKey as SodiumSecretKey,
+use ed25519_dalek::{
+    Keypair, PublicKey as DalekPublicKey, SecretKey as DalekPrivateKey,
+    Signature as DalekSignature, Signer as _, Verifier as _,
 };
 
-pub use sodiumoxide::crypto::sign::{
-    Signature, PUBLICKEYBYTES as PUBLIC_LENGTH,
-    SECRETKEYBYTES as SECRET_LENGTH, SIGNATUREBYTES as SIGNATURE_LENGTH,
+pub use ed25519_dalek::{
+    KEYPAIR_LENGTH as KEYPAIRBYTES, PUBLIC_KEY_LENGTH as PUBLICKEYBYTES,
+    SECRET_KEY_LENGTH as SECRETKEYBYTES,
```

Comment:
```suggestion
    SECRET_KEY_LENGTH as PRIVATEKEYBYTES,
```

And I propose to generally use "private" instead of "secret".
`rg SecretKey`
```
src/crypto/parse.rs
110:impl FromStr for exchange::SecretKey {
114:        use sodiumoxide::crypto::kx::{SecretKey, SECRETKEYBYTES};
120:            SecretKey::from_slice(bytes.as_slice()).context(UnexpectedSize)?;
147:        use ed25519_dalek::SecretKey;
152:        let sodium = SecretKey::from_bytes(bytes.as_slice()).context(Dalek)?;

src/crypto/sign.rs
9:    Keypair, PublicKey as DalekPublicKey, SecretKey as DalekPrivateKey,
168:    /// Get the `SecretKey` in this `KeyPair`
178:    /// Sign a message using this `SecretKey`

src/crypto/seal.rs
13:    PublicKey as SodiumPublicKey, SecretKey as SodiumSecretKey,
40:pub struct SecretKey(SodiumSecretKey);
46:    secret: SecretKey,
56:            secret: SecretKey(secret),
131:    /// Get a reference to the `SecretKey` in use within this `Seal`
132:    pub fn secret(&self) -> &SecretKey {

src/crypto/key/exchange.rs
12:    PublicKey as SodiumPubKey, SecretKey as SodiumSecKey,
49:impl fmt::Display for SecretKey {
73:/// A `SecretKey` used to compute a shared secret with a remote party
74:pub struct SecretKey(SodiumSecKey);
76:impl AsRef<[u8]> for SecretKey {
82:impl From<SodiumSecKey> for SecretKey {
92:    secret: SecretKey,
97:    pub fn new(secret: SecretKey, public: PublicKey) -> Self {
108:            secret: SecretKey(secret),
117:    /// Get the `SecretKey` from this `KeyPair`
118:    pub fn secret(&self) -> &SecretKey {
195:    /// shared secret using the given `SecretKey`
```